pkg/fleet-manager/application/rollout_helper.go

```diff
@@ -0,0 +1,459 @@
+		return ctrl.Result{}, errors.Errorf("Config Canary Deployment and A/B Testing or Blue/Green Deployment at the same time")
+	}
+
+	return ctrl.Result{}, nil
+}
+
+// Create an HPA based on the information provided by the workload
+func createHPAConfig(workload *applicationapi.CrossNamespaceObjectReference, deployment *appsv1.Deployment) autoscalingv1.HorizontalPodAutoscaler {
+	cpuUtilization := int32(99)
+
+	replica := int32(1)
+	if deployment.Spec.Replicas == nil {
+		deployment.Spec.Replicas = &replica
+	}
+
+	hpa := autoscalingv1.HorizontalPodAutoscaler{
+		TypeMeta: metav1.TypeMeta{
+			Kind:       "HorizontalPodAutoscaler",
+			APIVersion: "autoscaling/v2",
+		},
+		ObjectMeta: metav1.ObjectMeta{
+			Namespace: workload.Namespace,
+			Name:      workload.Name,
+		},
+		Spec: autoscalingv1.HorizontalPodAutoscalerSpec{
+			ScaleTargetRef: autoscalingv1.CrossVersionObjectReference{
+				APIVersion: "apps/v1",
+				Kind:       "Deployment",
+				Name:       workload.Name,
+			},
+			MinReplicas:                    deployment.Spec.Replicas,
+			MaxReplicas:                    *deployment.Spec.Replicas + 2,
+			TargetCPUUtilizationPercentage: &cpuUtilization,
+		},
+	}
+
+	return hpa
+}
+
+func installPrivateTestloader(ctx context.Context, namespacedName types.NamespacedName, fleetCluster fleetmanager.FleetCluster, labels map[string]string) (ctrl.Result, error) {
+	log := ctrl.LoggerFrom(ctx)
+
+	// Creating a private testload deployment from a configuration file
+	filepath := "./rollout-testloader"
+	deployCfg := "deployment.yaml"
+	deployFilepath := filepath + deployCfg
+	deploy, err1 := generateDeployConfig(ctx, deployFilepath, namespacedName.Name, namespacedName.Namespace)
+	if err1 != nil {
+		return ctrl.Result{}, fmt.Errorf("failed get testloader deployment configuration: %v", err1)
+	}
+	// Set up labels to make sure it's a resource created by kurator.
+	deploy.SetLabels(labels)
+	clusterClient := fleetCluster.Client.CtrlRuntimeClient()
+	if err := clusterClient.Create(ctx, deploy); err != nil {
+		return ctrl.Result{}, errors.Wrapf(err, "failed to create private testloader deployment")
+	}
+
+	// Creating a private testload service from a configuration file
+	svcCfg := "service.yaml"
+	svcFilepath := filepath + svcCfg
+	svc, err2 := generateSVCConfig(ctx, svcFilepath, namespacedName.Name, namespacedName.Namespace)
+	if err2 != nil {
+		return ctrl.Result{}, fmt.Errorf("failed get testloader service configuration: %v", err2)
+	}
+	// Set up labels to make sure it's a resource created by kurator.
+	svc.SetLabels(labels)
+	if err := clusterClient.Create(ctx, svc); err != nil {
+		return ctrl.Result{}, errors.Wrapf(err, "failed to create private testloader service")
+	}
+
+	log.Info("Create private workload successful")
+	return ctrl.Result{}, nil
+}
+
+func generateDeployConfig(ctx context.Context, filepath string, name, namespace string) (*appsv1.Deployment, error) {
+	log := ctrl.LoggerFrom(ctx)
+
+	file, err := os.Open(filepath)
```

Comment:
You can not read from it, since we didn't copy the yamls to image file system.

I forget how you installed global test loader

Reply:
Kurator will copy the configuration of a specific folder to image system.
I will then convert the relevant yaml to go.